explainer-server/app/models/ExplainerStore.scala

```diff
@@ -14,21 +14,41 @@ import scala.concurrent.Future
 import org.joda.time.DateTime
 import shared.util.ExplainerAtomImplicits
 
+import com.gu.pandomainauth.model.{User => PandaUser}
+
 class ExplainerStore @Inject() (config: Config) extends ExplainerAtomImplicits  {
 
   val explainerDB = new ExplainerDB(config)
 
-  def buildAtomWithDefaults(id: String, explainerAtom: ExplainerAtom, revision: Long): Atom = {
+  def buildAtomWithDefaults(id: String, explainerAtom: ExplainerAtom, contentChangeDetails: ContentChangeDetails, user: PandaUser): Atom = {
     Atom(
       id = id,
       atomType = AtomType.Explainer,
       defaultHtml = "",
       data = AtomData.Explainer(explainerAtom),
-      contentChangeDetails = ContentChangeDetails(lastModified = Some(ChangeRecord(DateTime.now.getMillis, user=None)), revision = revision)
+      contentChangeDetails = contentChangeDetails
+    )
+  }
+
+  def pandaUserToAtomUser(user: PandaUser): Option[User] = {
+    Some(User(user.email,Some(user.firstName),Some(user.lastName)))
+  }
+
+  def contentChangeDetailsBuilder(user: PandaUser, existingContentChangeDetails: Option[ContentChangeDetails], updateCreated: Boolean, updateLastModified: Boolean, updatePublished: Boolean): ContentChangeDetails = {
+    def buildChangeRecord(existingRecord: Option[ChangeRecord], shouldUpdate: Boolean) = {
+      if (shouldUpdate) {
+        Some(ChangeRecord(DateTime.now.getMillis, user=pandaUserToAtomUser(user)))
+      } else if (existingRecord.isDefined) existingRecord else None
+    }
+    ContentChangeDetails(
+      created      = buildChangeRecord(existingContentChangeDetails.flatMap(_.created)     , updateCreated),
+      lastModified = buildChangeRecord(existingContentChangeDetails.flatMap(_.lastModified), updateLastModified),
+      published    = buildChangeRecord(existingContentChangeDetails.flatMap(_.published)   , updatePublished),
+      revision     = existingContentChangeDetails.map(_.revision).getOrElse(1)

```

Comment:
when updating, we need to update this. maybe 

`existingContentChangeDetails.map(_.revision).getOrElse(0)+1`